lib/object_pool.js

```diff
@@ -178,4 +196,13 @@ ObjectPool.prototype.free = function free(inst) {
         this.freeList.push(inst);
     }
     this.outstanding -= 1;
+
+    var i;
+    if (ObjectPool.debug) {
+        for (i = 0; i < this.outstandingList.length; i++) {
+            if (this.outstandingList[i] === inst) {
+                this.outstandingList.splice(i, 1);
+            }
```

Comment:
No; objects are only added when they're allocated. An object would have to be freed twice, but we have an assertion guard against that. There could be a return after we do the splice. I'll add that in.